File: Data Structures & Algorithms/binary-tree-right-side-view/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        q = deque()
        
        ret = []
        if not root:
            return ret
        q.append(root)
            
        while q:
            lvLen = len(q)    

            filled = False
            for i in range(lvLen):
                node = q.popleft()

                if not node:
                    continue

                if not filled:
                    ret.append(node.val)
                    filled = True
                
                q.append(node.right)
                q.append(node.left)
        
        return ret
        

            
            

File: Data Structures & Algorithms/pow-x-n/submission-2.cpp
class Solution {
public:
    double myPow(double x, int n) {
        if (x == 0) {
            return 0;
        }

        if (n == 0) {
            return 1;
        }

        long sqr = abs((long)n);
        double ret = 1;

        while (sqr) {
            if (sqr & 1) {
                ret *= x;
            }
            x *= x;
            sqr >>= 1; 
        }

        return n > 0 ? ret : (1 / ret);
    }
};
